Update SLEAP guide for newest module version (#65)
neuroinformatics-unit/HowTo: docs/source/data_analysis/HPC-module-SLEAP.md

| [SWC](https://www.sainsburywellcome.org/web/)                   | Sainsbury Wellcome Centre                    |
| [HPC](https://en.wikipedia.org/wiki/High-performance_computing) | High Performance Computing                   |
| [IT](https://en.wikipedia.org/wiki/Information_technology)      | Information Technology                       |
| [GUI](https://en.wikipedia.org/wiki/Graphical_user_interface)   | Graphical User Interface                     |
| [SLURM](https://slurm.schedmd.com/)                             | Simple Linux Utility for Resource Management |

## Prerequisites

::: {dropdown} Note on managed Linux desktops
:color: info
:icon: info

The SWC's IT team offers managed desktop computers equipped with a Linux image. These machines are already part of SWC's trusted domain and have direct access to SLURM, the HPC modules, and the SWC filesystem.

If you have access to one of these desktops,
you can skip the pre-requisite steps.
You may simply open a terminal, type `module load SLEAP`,
and start using SLEAP directly, as you would on any local
Linux machine. All SLEAP commands should work as expected,
including `sleap-label` for launching the GUI.

> That said, you may still want to offload GPU-intensive tasks to other HPC nodes (e.g. because the desktop's GPU is not powerful enough or because you need to run many jobs in parallel). In that case, you may

Reviewer: ```suggestion
That said, you may still want to offload GPU-intensive tasks to an HPC nodes (e.g. because the desktop's GPU is not powerful enough or because you need to run many jobs in parallel). In that case, you may
```
(since their workstations are not HPC nodes right?)

:::{tab-item} Windows and Linux
```{code-block} console

> $ conda create -y -n sleap -c conda-forge -c nvidia -c sleap -c anaconda sleap=1.3.3

Reviewer: It feels a bit of an unnecessary burden to ensure that this section is in sync with SLEAP's docs.

I think we can remove most of it and keep until: 
> We recommend following the official [SLEAP installation guide](https://sleap.ai/installation.html).

I would keep your clarification at the end, I like it, but maybe rephrase a bit to include why:
> To minimise the risk of issues due to incompatibilities between versions, ensure the version of your local installation of SLEAP matches the one you plan to load in the cluster.

Author: agreed, you're totally right.

```
:::

(sleap-training)=
## Model training

> This will consist of two parts - [preparing a training job](prepare-the-training-job)

Reviewer: ```suggestion
This will consist of two parts: [preparing a training job](prepare-the-training-job)
```

- If requesting more memory doesn't help, you can try reducing the size of your SLEAP models. You may tweak the model backbone architecture, or play with *Input scaling*, *Max stride* and *Batch size*. See SLEAP's [documentation](https://sleap.ai/) and [discussion forum](https://github.com/talmolab/sleap/discussions) for more details.
```

(evaluate-the-trained-models)=

> ### Evaluate the trained models

Reviewer: ```suggestion
## Model evaluation
```
maybe? That way it matches a bit better the train, eval, inference scheme.

If you encounter troubles with using the SLEAP module, contact
Niko Sirmpilatze of the SWC [Neuroinformatics Unit](https://neuroinformatics.dev/).

To completely exit the HPC cluster, you will need to type `exit` or

> `logout` until you are back to your local machine's terminal prompt.

Reviewer: ```suggestion
`logout` until you are back to the terminal prompt of your local machine.
```
